src/utils/Timezone.php

```diff
@@ -105,40 +108,40 @@ public static function init() : void{
 	public static function detectSystemTimezone() : string|false{
 		switch(Utils::getOS()){
 			case Utils::OS_WINDOWS:
-				$regex = '/(UTC)(\+*\-*\d*\d*\:*\d*\d*)/';
+				$keyPath = 'HKLM\\SYSTEM\\CurrentControlSet\\Control\\TimeZoneInformation';
 
 				/*
-				 * wmic timezone get Caption
-				 * Get the timezone offset
-				 *
-				 * Sample Output var_dump
-				 * array(3) {
-				 *	  [0] =>
-				 *	  string(7) "Caption"
-				 *	  [1] =>
-				 *	  string(20) "(UTC+09:30) Adelaide"
-				 *	  [2] =>
-				 *	  string(0) ""
-				 *	}
+				 * Get the timezone offset through the registry
 				 */
-				exec("wmic timezone get Caption", $output);
-
-				$string = trim(implode("\n", $output));
-
-				//Detect the Time Zone string
-				preg_match($regex, $string, $matches);
-
-				if(!isset($matches[2])){
-					return false;
-				}
-
-				$offset = $matches[2];
-
-				if($offset === ""){
-					return "UTC";
+				exec("reg query " . escapeshellarg($keyPath), $output);
+
+				foreach($output as $line){
+					if(preg_match('/ActiveTimeBias\s+REG_DWORD\s+0x+([0-9a-fA-F]+)/', $line, $matches) > 0){
```

Comment:
```suggestion
					if(preg_match('/ActiveTimeBias\s+REG_DWORD\s+0x([0-9a-fA-F]+)/', $line, $matches) > 0){
```

Just by eyeballing it, the extra plus doesn't seem right to me. Though it depends on the output. A sample for future readers would be nice.